internal/server/handlers/rbac/monitoring_instance_test.go

```diff
@@ -0,0 +1,743 @@
+				t.Parallel()
+				k8sMock := newConfigMapMock(tc.policy)
+				enf, err := rbac.NewEnforcer(ctx, k8sMock, zap.NewNop().Sugar())
+				require.NoError(t, err)
+
+				next := next()
+
+				h := &rbacHandler{
+					next:       next,
+					log:        zap.NewNop().Sugar(),
+					enforcer:   enf,
+					userGetter: testUserGetter,
+				}
+				_, err = h.GetMonitoringInstance(ctx, "default", "monitoring-instance-1")
+				assert.ErrorIs(t, err, tc.wantErr)
+			})
+		}
+	})
+
+	t.Run("CreateMonitoringInstance", func(t *testing.T) {
+		t.Parallel()
+		next := func() *handlers.MockHandler {
+			next := handlers.MockHandler{}
+			next.On("CreateMonitoringInstance", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(
+				&everestv1alpha1.MonitoringConfig{}, nil,
+			)
+			return &next
+		}
+
+		testCases := []struct {
+			desc    string
+			policy  string
+			wantErr error
+		}{
+			{
+				desc: "admin",
+				policy: newPolicy(
+					"g, bob, role:admin",
+				),
+			},
+			{
+				desc: "all actions for all monitoring-instances in all namespaces",
+				policy: newPolicy(
+					"p, role:test, monitoring-instances, *, */*",
+					"g, bob, role:test",
+				),
+			},
+			{
+				desc: "all actions for all monitoring-instances in default namespace",
+				policy: newPolicy(
+					"p, role:test, monitoring-instances, *, default/*",
+					"g, bob, role:test",
+				),
+			},
+			{
+				desc: "all actions for 'monitoring-instance-1' in default namespace",
+				policy: newPolicy(
+					"p, role:test, monitoring-instances, *, default/monitoring-instance-1",
+					"g, bob, role:test",
+				),
+			},
+			{
+				desc: "create only action for all monitoring-instances in all namespaces",
+				policy: newPolicy(
+					"p, role:test, monitoring-instances, create, */*",
+					"g, bob, role:test",
+				),
+			},
+			{
+				desc: "create only action for all monitoring-instances in default namespace",
+				policy: newPolicy(
+					"p, role:test, monitoring-instances, create, default/*",
+					"g, bob, role:test",
+				),
+			},
+			{
+				desc: "create only action for 'monitoring-instance-1' in default namespace",
+				policy: newPolicy(
+					"p, role:test, monitoring-instances, create, default/monitoring-instance-1",
+					"g, bob, role:test",
+				),
+			},
+			{
+				desc: "update only action for all monitoring-instances in all namespaces",
+				policy: newPolicy(
+					"p, role:test, monitoring-instances, update, */*",
+					"g, bob, role:test",
+				),
+				wantErr: ErrInsufficientPermissions,
+			},
+			{
+				desc: "update only action for all monitoring-instances in default namespace",
+				policy: newPolicy(
+					"p, role:test, monitoring-instances, update, default/*",
+					"g, bob, role:test",
+				),
+				wantErr: ErrInsufficientPermissions,
+			},
+			{
+				desc: "update only action for some monitoring-instances in default namespace",
+				policy: newPolicy(
+					"p, role:test, monitoring-instances, update, default/some",
```

Comment:
Commenting here in the `update` but the same is true in the `read` and `delete`. Shouldn't this match the name of the resource (`monitoring-instance-1`) to make sure that `update` isn't a super-set of `create`? By using `some` we are not validating that `update default/monitoring-instance-1` doesn't lead to `create` as well which I think was the purpose of iterating through `read`, `update`, `delete`.

Reply:
Right, yeah, makes sense.

I think I might have intended to do `create default/some` - to test that having `create` for some, does not lead to giving permissions for another